backend/src/xfd_django/xfd_api/tests/test_check_user_expiration.py

```diff
@@ -0,0 +1,222 @@
+"""Tests for checkUserExpiration Lambda task."""
+# Standard Python Libraries
+from datetime import timedelta
+
+# Third-Party Libraries
+from django.utils import timezone
+import pytest
+from xfd_api.tasks import checkUserExpiration
+from xfd_mini_dl.models import User
+
+
+@pytest.mark.django_db(transaction=True, databases=["default", "mini_data_lake"])
+def test_check_user_expiration_sends_emails_and_deletes(monkeypatch):
+    """Users in each inactivity band get the right email behavior and 90+ day users are deleted."""
+    # Freeze "now" in the task module
+    frozen_now = timezone.now()
+
+    # Patch the now() used inside checkUserExpiration
+    monkeypatch.setattr(checkUserExpiration, "now", lambda: frozen_now)
+
+    # Capture emails instead of actually sending
+    emails_sent = []
+
+    def fake_send_email(to_addr, subject, body):
+        emails_sent.append({"to": to_addr, "subject": subject, "body": body})
+
+    monkeypatch.setattr(checkUserExpiration, "send_email", fake_send_email)
+
+    # Create users in different inactivity windows
+    # 20 days inactive -> should get NO email
+    user_20 = User.objects.create(
+        first_name="Twenty",
+        last_name="Days",
+        email="20@example.com",
+        last_logged_in=frozen_now - timedelta(days=20),
+    )
+
+    # 35 days inactive -> 30-day notice
+    user_35 = User.objects.create(
+        first_name="ThirtyFive",
+        last_name="Days",
+        email="35@example.com",
+        last_logged_in=frozen_now - timedelta(days=35),
+    )
+
+    # 60 days inactive -> 45-day notice
+    user_60 = User.objects.create(
+        first_name="Sixty",
+        last_name="Days",
+        email="60@example.com",
+        last_logged_in=frozen_now - timedelta(days=60),
+    )
+
+    # 120 days inactive -> 90-day notice + deletion
+    user_120 = User.objects.create(
+        first_name="OneTwenty",
+        last_name="Days",
+        email="120@example.com",
+        last_logged_in=frozen_now - timedelta(days=120),
+    )
+
+    # Run the task
+    checkUserExpiration.check_user_expiration()
```

Comment:
When this gets deployed to staging, it will be reading from the staging database and will send to all users present. Do we want to target just the confined users that you state at 65-71, that link back to the users above? If so we will want to have a default value of target email as a parameter at the check_user_expiration(target_email=None) function and modify your.